src/Illuminate/Support/Arr.php

```diff
@@ -62,19 +63,24 @@ public static function build($array, callable $callback)
     /**
      * Collapse an array of arrays into a single array.
      *
-     * @param  array  $array
+     * @param  \Traversable|array  $array
      * @return array
      */
     public static function collapse($array)
     {
+        // Iterate on the array for performance.
```

Comment:
I test it on 1000000 items in an array.

In php5, pure iterate over array vs pure iterate over iterator can lead to 0.043 vs 0.238
But in php7, it lead to 0.031 vs 0.081

Yes, it needed, and the code is more readable